### PR TITLE
Support async/await in Protocol with Associated Type; closes #441

### LINKED ISSUE #441: async/await is not fully supported in protocol with associated types
Although there is a merged PR here related to `async/await` #410, the feature is not fully supported, at least, in protocols with associated types.

**Example:**

```
protocol Datasource<T>{
    associatedtype T
    func get() async -> Result<T, Error>
}
```

**Snippet from generated class:**

Screen Shot 2022-12-30 at 21 03 18

**IDE complains saying:**

> Invalid conversion from 'async' function of type '() async -> Result<T, any Error>' to synchronous function type '() -> Result<T, any Error>'

**Solution:**
I have added required keywords to generated code, it is fixed:

Screen Shot 2022-12-30 at 21 03 40

### PR DESCRIPTION
**Pull request by mecoFarid**

### CODE REVIEW

File: Tests/Swift/GenericProtocolTest.swift [hunk 1]
```diff
@@ -34,8 +50,8 @@ private class GenericProtocolConformerClass<C: AnyObject, V>: GenericProtocol {
             return 0
         }
     }
-
-    func compute(classy: C, value: V) -> C {
+
```
Comment (MatyasKriz) [inline Tests/Swift/GenericProtocolTest.swift:53]: Please remove the whitespace here. 🙂

File: Tests/Swift/GenericProtocolTest.swift [hunk 2]
```diff
@@ -196,4 +215,37 @@ class GenericProtocolTest: XCTestCase {
 
         verify(mock, times(2)).readWritePropertyV.get()
     }
+    
+    func testComputeAsync() async{
```
Comment (MatyasKriz) [inline Tests/Swift/GenericProtocolTest.swift:219]: Please add a space before the opening brace. I think I saw it 3 times here.

**Review by MatyasKriz — APPROVED** [review]
Thanks for the PR, it looks solid. 🙂 I'll take another look after work and merge it.

Thank you for including the tests.

File: Tests/Swift/GenericProtocolTest.swift (hunk 2 above)
Comment (mecoFarid) [reply Tests/Swift/GenericProtocolTest.swift]: Done

**Comment by mecoFarid** [comment]
@MatyasKriz As you have noticed, I did not include all the methods with `async/await` functionality, I just decided to modify their signature and make it a bit shorter by adding only two new methods for `async/await` functionality. To me they look quite enough but if you have specific coding convention that I am unaware of just let me know

**Comment by MatyasKriz** [comment]
Sure, makes sense to me. 🙂 I just didn't get to it yesterday, merged. Thank you for your contribution!

**Pull request closed without merging**